test/unit/space_index_test.lua

```diff
@@ -0,0 +1,301 @@
+            case.fun(space, space.index[index_name])
+            space:insert({1, "1"})
+
+            local stderr = ""
+            helpers.retrying({}, function()
+                stderr = stderr .. capture:flush().stderr
+                t.assert_str_contains(stderr, "expiration: stop task")
+            end)
+        end)
+
+        t.assert_equals(helpers.iteration_result, {})
+        t.assert_equals(#expirationd.tasks(), 1)
+        t.assert_not_equals(expirationd.task(task_name), nil)
+        t.assert_equals(task:statistics().restarts, 1)
+
+        task:kill()
+    end
+    g["test_not_stop_after_" .. name .. "if_id"] = function(cg)
+        local task_name = "clean_all"
+        local space_id = cg.space.id
+        local index_id = 1
+
+        helpers.iteration_result = {}
+        local task = expirationd.start(task_name, space_id, helpers.is_expired_debug,
+                                       {index = index_id})
+        helpers.retrying({}, function()
+            t.assert_equals(helpers.iteration_result, {{1, "1"}})
+        end)
+        helpers.iteration_result = {}
+
+        local space = box.space[space_id]
+        case.fun(space, space.index[index_id])
+        space:insert({1, "1"})
+
+        helpers.retrying({}, function()
+            t.assert_equals(helpers.iteration_result, {{1, "1"}})
+        end)
+        t.assert_equals(#expirationd.tasks(), 1)
+        t.assert_not_equals(expirationd.task(task_name), nil)
+        t.assert_equals(task:statistics().restarts, 1)
+
+        task:kill()
+    end
+end
+
+local drop_cases = {
+    index_drop = {
+        fun = function(_, index)
+            index:drop()
+        end,
+    },
+    space_drop = {
+        fun = function(space, _)
+            space:drop()
+        end,
+    }
+}
+
+for name, case in pairs(drop_cases) do
+    g["test_stop_after_" .. name] = function(cg)
+        local task_name = "clean_all"
+        local index_id = 1
+
+        helpers.iteration_result = {}
+        local task = expirationd.start(task_name, cg.space.id, helpers.is_expired_debug,
+                                       {index = index_id})
+        helpers.retrying({}, function()
+            t.assert_equals(helpers.iteration_result, {{1, "1"}})
+        end)
+        helpers.iteration_result = {}
+
+        local capture = luatest_capture:new()
+        capture:wrap(true, function()
+            case.fun(cg.space, cg.space.index[index_id])
+
+            local stderr = ""
+            helpers.retrying({}, function()
+                stderr = stderr .. capture:flush().stderr
+                t.assert_str_contains(stderr, "expiration: stop task")
+            end)
+        end)
+
+        t.assert_equals(helpers.iteration_result, {})
+        t.assert_equals(#expirationd.tasks(), 1)
+        t.assert_not_equals(expirationd.task(task_name), nil)
+        t.assert_equals(task:statistics().restarts, 1)
+
+        task:kill()
+    end
+end
+
+function g.test_run_after_truncate(cg)
```

Comment:
Why do we test this here? What possibly could change?

Reply:
And we tests that all fine after truncate. For my taste, it is ok  here.

Reply:
But rename is drop+create, not a truncate

Reply:
Yes, truncate is mentioned in the comments. create/drop/truncate/rename are a space manipulations with possible side-effects in the future.

Therefore, we need to add at least a basic test for truncate. Even if we are sure that now it will not lead to problems.

Reply:
Well, ok, let it be